src/index.rs

```diff
@@ -515,17 +514,14 @@ fn discover_paths<P: AsRef<Path>>(
         .collect()
 }
 
-fn scan_entry(
-    path: &CanonicalPath,
-    metadata: Metadata,
-) -> Result<IndexEntry, Error> {
+fn scan_entry(path: &CanonicalPath, metadata: Metadata) -> Result<IndexEntry> {
     if metadata.is_dir() {
-        return Err(Error::msg("DirEntry is directory"));
+        return Err(ArklibError::Path);
```

Comment:
Detailed message would be good unless that's the only `ArklibError::Path`  in the whole codebase.

Reply:
I have added a String message on the Path error